src/xrGame/DiscordRPC.cpp

```diff
@@ -0,0 +1,53 @@
+#include "stdafx.h"
+#include "DiscordRPC.h"
+#include <DiscordRPC/discord_register.h>
+#include <DiscordRPC/discord_rpc.h>
+#include <xrCore/Text/StringConversion.hpp>
+
+DiscordRPC g_DiscordRPC;
+const pcstr g_discordAppID = "869824972464488510";
+
+void DiscordRPC::Init()
+{
+    DiscordEventHandlers handle{};
+    Discord_Initialize(g_discordAppID, &handle, 1, nullptr);
+
+    xr_sprintf(build_name, "build %u", Core.GetBuildId());
+}
+
+void DiscordRPC::Update(pcstr level_name)
+{
+    DiscordRichPresence discordPresence{};
+
+    discordPresence.startTimestamp      = start_time;
+    discordPresence.smallImageText      = build_name;
+    discordPresence.largeImageKey       = "current_level_name";
+    discordPresence.smallImageKey       = "build_name";
+
+    if (level_name)
+    {
+        std::locale myLocale("");
+        xr_string temp = StringToUTF8(level_name, myLocale);
+        xr_sprintf(current_level_name, "%s", temp);
+    }
+
+    discordPresence.state               = current_level_name;
+    discordPresence.largeImageText      = current_level_name;
+    discordPresence.details             = current_task_name;
```

Comment:
@Xottab-DUTY should we leave such `column` style for assignment operators?

Reply:
Let's leave it as is, yes.